### ISSUE TITLE
dynamic data includes

### ISSUE THREAD

**Issue by waynedpj** [assemble/assemble#432]
ahoy,

before i try writing a plugin to do this, i am wondering if there is a easier way to dynamically include certain data files for the current page, specifically any data files from the current page back to the root of the project?

take this example directory hierarchy:

```
index.html
releases
    ├── index.html
    ├── A
    │   ├── index.html
    │   ├── info.yaml
    │   └── tracks
    │       ├── track-01
    │       │   ├── index.html
    │       │   └── lyrics.yaml
    │       ├── track-02
    │       │   ├── index.html
    │       │   └── lyrics.yaml
    │       └── track-03
    │           ├── index.html
    │           └── lyrics.yaml
    └── B
        ├── index.html
        ├── info.yaml
        └── tracks
            ├── track-01
            │   ├── index.html
            │   └── lyrics.yaml
            └── track-02
                ├── index.html
                └── lyrics.yaml
```

i would like to have each track's `lyrics.yaml` and it's parent release's `info.yaml` data files loaded into context only while processing that particular track. so in this example, while processing each track's `index.html`:
- the `lyrics` property would only contain the data from the `lyrics.yaml` file located in the same directory as the `index.html`
- the `info` property would only contain the data from the track's parent release's `info.yaml`

basically any data files from the currently processed page's directory up to the project root would be included in the context, making a type of cascade where the context follows the directory structure to the currently processed file.

and while right now i can choose the data files for each target, that would mean that i would have to create a collection for each track and give it a custom `data` option value.

i am slowly learning how flexible Assemble is, so i imagine i am missing an easier solution already available.

thanks in advance.
